### PR TITLE
Prefer constant division over multiplication for length checks

### PR DESCRIPTION
**Pull request by akallabeth**
@hardening thanks for pointing out we´ve still had this.

### CODE REVIEW

File: channels/urbdrc/client/data_transfer.c [hunk 1]
```diff
@@ -801,7 +801,7 @@ static UINT urb_isoch_transfer(IUDEVICE* pdev, URBDRC_CHANNEL_CALLBACK* callback
 	Stream_Read_UINT32(s, NumberOfPackets); /** NumberOfPackets */
 	Stream_Read_UINT32(s, ErrorCount);      /** ErrorCount */
 
-	if (Stream_GetRemainingLength(s) < NumberOfPackets * 12 + 4)
+	if (Stream_GetRemainingLength(s) < NumberOfPackets * 12ULL + 4ULL)
 		return ERROR_INVALID_DATA;
```
Comment (hardening) [inline channels/urbdrc/client/data_transfer.c:805]: AFAICT this one does not address the possible overflow problem

File: libfreerdp/core/update.c [hunk 2]
```diff
@@ -2122,7 +2122,7 @@ BOOL update_read_refresh_rect(rdpUpdate* update, wStream* s)
 	Stream_Read_UINT8(s, numberOfAreas);
 	Stream_Seek(s, 3); /* pad3Octects */
 
-	if (Stream_GetRemainingLength(s) < ((size_t)numberOfAreas * 4 * 2))
+	if (Stream_GetRemainingLength(s) / 2 / 4 < numberOfAreas)
 		return FALSE;
```
Comment (hardening) [inline libfreerdp/core/update.c:2126]: `/ 8` would be more explicit, I had to re-read it twice to be sure it was correct

Comment (akallabeth) [reply libfreerdp/core/update.c:2126]: kept it the same as it was before (e.g. 4 coordinates, 2 bytes size for each)

File: channels/urbdrc/client/data_transfer.c (hunk 1 above)
Comment (akallabeth) [reply channels/urbdrc/client/data_transfer.c:805]: it does, as it forces the data type to be `uint64`. `NumberOfPackets` is `UINT32`

File: libfreerdp/core/orders.c [hunk 3]
```diff
@@ -2664,7 +2664,7 @@ static BOOL update_decompress_brush(wStream* s, BYTE* output, size_t outSize, BY
 	const BYTE* palette = Stream_Pointer(s) + 16;
 	const size_t bytesPerPixel = ((bpp + 1) / 8);
 
-	if (Stream_GetRemainingLength(s) < 16 + bytesPerPixel * 4)
+	if (Stream_GetRemainingLength(s) < 16ULL + bytesPerPixel * 4ULL)
 		return FALSE;
```
Comment (hardening) [inline libfreerdp/core/orders.c:2668]: Like above to address all the overflow cases, we should probably do a 
`if ((Stream_GetRemainingLength(s) < 16ULL) || ((Stream_GetRemainingLength(s) -16) / 4 < bytesPerPixel)`

**Review by hardening — CHANGES_REQUESTED** [review]
Except my remarks LGTM

File: libfreerdp/core/orders.c (hunk 3 above)
Comment (akallabeth) [reply libfreerdp/core/orders.c:2668]: not required due to range of input data type. `bpp` is (at most) `UINT32` and we operate on `UINT64` here.

Comment (akallabeth) [reply libfreerdp/core/orders.c:2668]: sorry, even more narrow, at most `UINT8`

File: libfreerdp/core/update.c (hunk 2 above)
Comment (akallabeth) [reply libfreerdp/core/update.c:2126]: done.

**Pull request merged**